recipes/libtorch/all/conanfile.py

```diff
@@ -0,0 +1,667 @@
+            del self.options.with_qnnpack
+        if not is_apple_os(self):
+            self.options.rm_safe("with_metal")
+            self.options.rm_safe("with_magma")
+            self.options.rm_safe("with_mps")
+        if self.settings.os != "Linux":
+            del self.options.with_numa
+        if self.settings.os != "Android":
+            self.options.rm_safe("with_nnapi")
+            self.options.rm_safe("with_snpe")
+            self.options.with_vulkan = False
+        if self._is_mobile_os:
+            self.options.blas = "eigen"
+            self.options.build_lazy_ts_backend = False
+            del self.options.with_fbgemm
+            # del self.options.distributed
+        if self.settings.arch not in ["x86", "x86_64", "armv8"]:
+            self.options.rm_safe("with_mkldnn")
+        if not is_apple_os(self) or self.settings.os not in ["Linux", "Android"]:
+            del self.options.with_nnpack
+        self.options.with_itt = self.settings.arch in ["x86", "x86_64"]
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        if not self.options.get_safe("with_cuda"):
+            self.options.rm_safe("build_lazy_cuda_linalg")
+            self.options.rm_safe("debug_cuda")
+            self.options.rm_safe("with_cudnn")
+            self.options.rm_safe("with_cusparselt")
+            self.options.rm_safe("with_nvrtc")
+            if not self.options.get_safe("with_rocm"):
+                self.options.rm_safe("with_nccl")
+        if not self.options.with_vulkan:
+            self.options.rm_safe("vulkan_fp16_inference")
+            self.options.rm_safe("vulkan_relaxed_precision")
+        if not self.options.get_safe("fbgemm"):
+            self.options.rm_safe("fakelowp")
+
+        # numa static can't be linked into shared libs.
+        # Because Caffe2_detectron_ops* libs are always shared, we have to force
+        # libnuma shared even if libtorch:shared=False
+        if self.options.get_safe("with_numa"):
+            self.options["libnuma"].shared = True
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    @property
+    def _depends_on_sleef(self):
+        return not self._is_mobile_os and not self.settings.os == "Emscripten"
+
+    @property
+    def _depends_on_pthreadpool(self):
+        return self._is_mobile_os or self._use_nnpack_family
+
+    @property
+    def _depends_on_flatbuffers(self):
+        return not self._is_mobile_os
+
+    @property
+    def _blas_cmake_option_value(self):
+        return {
+            "eigen": "Eigen",
+            "atlas": "ATLAS",
+            "openblas": "OpenBLAS",
+            "mkl": "MKL",
+            "veclib": "vecLib",
+            "flame": "FLAME",
+            "generic": "Generic"
+        }[str(self.options.blas)]
+
+    @property
+    def _use_nnpack_family(self):
+        return any(self.options.get_safe(f"with_{name}") for name in ["nnpack", "qnnpack", "xnnpack"])
```

Comment:
`with_xnnpack` can not be deleted because it is unsafe used on line 284 `if self.options.with_xnnpack:`

Reply:
Thanks for the comment, but I don't see `with_xnnpack` option being deleted anywhere? This specific line is querying the value, not removing the option.

Reply:
Sorry for the missing context. with_xnnpack must be deleted for MacOS build since this is not yet supported under Mac. For that cases it might be  problem.
I managed to build libtorch under linux and MacOS ARM with some modifications based on @valgur's work.
If you are interested you can see the actual verison:
https://github.com/joda01/imagec-recipes/actions/runs/12983296316

Reply:
Thanks! I'll definitely take a look.